### PR TITLE
Fixes #35263 - Index ModuleStreamErratumPackages properly

### PR DESCRIPTION
**Pull request by ianballou**
#### What are the changes introduced in this pull request?
Indexes ModuleStreamErratumPackages for all duplicate errata. Also includes a migration to reindex related library repositories.

#### Considerations taken when implementing this change?
It can be hard to tell which repositories exactly are affected by bug, so I reindex all repositories that have both duplicated errata and module streams.

#### What are the testing steps for this pull request?
1) Sync AppStream
2) Find the module stream with version 8050020210712211742, name httpd, stream 2.4
3) Find the erratum with ID RHSA-2021:4257
4) Duplicate the erratum with a dummy pulp_id
5) Move the all related ErratumPackages to the duplicate:
```ruby
erratum_packages = ::Katello::ErratumPackage.find(Katello::ModuleStreamErratumPackage.joins(:erratum_package => {:erratum => :repository_errata}).where("#{::Katello::RepositoryErratum.table_name}.repository_id" => appstream_repo_id).where("#{Katello::ModuleStreamErratumPackage.table_name}.module_stream_id = module_stream_id_from_above").pluck(:erratum_package_id))
erratum_packages.each {|ep| ep.update(erratum_id: duplicate_erratum.id)}
```
6) Check that appstream_repo.module_streams_without_errata returns the module stream mentioned above
7) Run appstream_repo.index_content
8) Note that appstream_repo.module_streams_without_errata still returns the module stream mentioned above
9) Patch in my changes
10) Run index_content again (in the PR: run the migration)
11) See that appstream_repo.module_streams_without_errata no longer returns the module stream mentioned above

### CODE REVIEW

**Comment by sjha4** [comment]
Without patch:

image

**Comment by sjha4** [comment]
Indexing after patch: 
image

**Comment by sjha4** [comment]
Works as expected. One question. Will ack once a decision is made around the question.

File: db/migrate/20220722193815_clean_missing_module_stream_erratum_packages.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+      if repo.module_streams.present? 
+        repo.index_content
+      end
+    end
+  end
+
+  def down
+    fail ActiveRecord::IrreversibleMigration
+  end
+end
```
[the remark this answers is not in the document]
Comment (ianballou) [reply db/migrate/20220722193815_clean_missing_module_stream_erratum_packages.rb:15]: My worry about that is that errata work differently on Pulp 2. I think when reindexing time comes around the duplicate errata may get created again.  The duplicate errata do exist in Pulp as well.

**Review by sjha4 — APPROVED** [review]

**Comment by ianballou** [comment]
Having API requests to backend services in a migration isn't a best practice, so we're going to move it into a rake task.

**Comment by ianballou** [comment]
I moved to a rake script. The new way to run is `rake katello:upgrades:3.18:create_missing_module_stream_erratum_packages`.  Making it an upgrade seems to make sense, but I'm not sure if these upgrades would get run automatically. I'm hoping that foreman-maintain would do it?

**Comment by ianballou** [comment]
I verified that this will run with `rake upgrade:run on Katello 3.18`.

**Comment by ianballou** [comment]
@sjha4 I'll just need a re-review since I moved away from the migration.

**Review by sjha4 — APPROVED** [review]
Looks good. 👍🏼

**Pull request merged**